### PR TITLE
Make pad2d support for variable paddings.

### LINKED ISSUE #14641: enhence pad op
pad op支持paddings作为variable输出。

### PR DESCRIPTION
**Pull request by wanghaoshuang**
fix #14641

### CODE REVIEW

File: paddle/fluid/operators/pad2d_op.cc [hunk 1]
```diff
@@ -319,20 +319,46 @@ void Pad2DGradEdgeNHWC(T* d_in_data, const int num, const int channels,
   }
 }
 
+inline void get_paddings(int* paddings,
```
Comment (qingqing01) [inline paddle/fluid/operators/pad2d_op.cc]:
```c++
static inline GetPaddings
```

File: paddle/fluid/operators/pad2d_op.cu [hunk 2]
```diff
@@ -287,20 +287,49 @@ __global__ void Pad2DGradEdgeNHWC(const int out_size, T* d_in_data,
   }
 }
 
+void get_paddings(int* paddings, const framework::ExecutionContext& context) {
```
Comment (qingqing01) [inline paddle/fluid/operators/pad2d_op.cu]:
```
static inline void GetPaddings() {
}
```

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -6917,16 +6917,17 @@ def pad2d(input,
     helper = LayerHelper('pad2d', **locals())
     dtype = helper.input_dtype(input_param_name='input')
     out = helper.create_variable_for_type_inference(dtype)
+    inputs = {'X': input}
+    attrs = {'mode': mode, 'pad_value': pad_value, 'data_frmat': data_format}
```
Comment (jerrywgz) [inline python/paddle/fluid/layers/nn.py]: data_format

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -6892,7 +6892,7 @@ def pad2d(input,
 
     Args:
         input (Variable): The input image with [N, C, H, W] format or [N, H, W, C] format.
-        paddings (tuple|list): The padding size. If padding is a tuple, it must
+        paddings (tuple|list|variable): The padding size. If padding is a tuple, it must
```
Comment (heavengate) [inline python/paddle/fluid/layers/nn.py]: variable -> Variable

File: paddle/fluid/operators/pad2d_op.cc (hunk 1 above)
Comment (wanghaoshuang) [reply paddle/fluid/operators/pad2d_op.cc]: Done.

File: paddle/fluid/operators/pad2d_op.cu (hunk 2 above)
Comment (wanghaoshuang) [reply paddle/fluid/operators/pad2d_op.cu]: Done.

File: python/paddle/fluid/layers/nn.py (hunk 3 above)
Comment (wanghaoshuang) [reply python/paddle/fluid/layers/nn.py]: Done.

File: python/paddle/fluid/layers/nn.py (hunk 4 above)
Comment (wanghaoshuang) [reply python/paddle/fluid/layers/nn.py]: Done.

**Pull request merged**